elf_binfmt.c

```diff
@@ -0,0 +1,93 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2023, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+#include <uk/assert.h>
+#include <uk/arch/ctx.h>
+#include <uk/binfmt.h>
+#include <uk/essentials.h>
+#include <uk/errptr.h>
+#include <uk/init.h>
+
+#if CONFIG_LIBUKSWRAND
+#include <uk/swrand.h>
+#endif /* CONFIG_LIBUKSWRAND */
+
+#include "elf_prog.h"
+
+static int uk_binfmt_load_elf(struct uk_binfmt_loader_args *args)
+{
+	struct elf_prog *prog;
+	__u64 rand[2];
+	int rc;
+
+	UK_ASSERT(args);
+	UK_ASSERT(args->alloc);
+
+	prog = elf_load_vfs(args->alloc, args->pathname, args->progname);
+	if (PTRISERR(prog) || !prog) {
```

Comment:
Will add a warning if not handled by the ELF handler, and an error otherwise. Will also remove the `!prog` check, not sure why I added that in the first place as from what I see `elf_load_vfs()` always returns < 0 for errors.